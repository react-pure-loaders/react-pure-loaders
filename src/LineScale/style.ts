import { keyframes } from '@emotion/react';
import { globalAnimation, lines } from '../mixins';
import delay from '../utilities';
import { Grid } from '../types';

const ballBeat = keyframes({
  '0%': {
    transform: 'scaley(1.0)',
  },
  '100%': {
    transform: 'scaley(1.0)',
  },
  '50%': {
    transform: 'scaley(0.4)',
  },
});

function lineScale() {
  const n = 5;
  let i = 1;
  const grid: Grid = {};
  for (; i < 10; i += 1) {
    grid[`> div:nth-of-type(${i})`] = {
      animation: `${ballBeat} 1s ${delay('0.1s', n, i)} infinite cubic-bezier(.2,.68,.18,1.08)`,
    };
  }

  return grid;
}

export default (color?: string) => ({
  ...lineScale(),
  ' > div': {
    ...lines(color),
    ...globalAnimation(),
    display: 'inline-block',
  },
});
