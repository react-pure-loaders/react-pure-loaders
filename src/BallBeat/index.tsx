/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
