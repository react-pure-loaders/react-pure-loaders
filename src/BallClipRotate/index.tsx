/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
