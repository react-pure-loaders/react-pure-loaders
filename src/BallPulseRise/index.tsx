/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
