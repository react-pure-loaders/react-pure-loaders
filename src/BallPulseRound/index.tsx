/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
