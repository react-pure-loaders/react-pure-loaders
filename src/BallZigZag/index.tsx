/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
