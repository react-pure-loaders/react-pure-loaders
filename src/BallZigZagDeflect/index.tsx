/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
