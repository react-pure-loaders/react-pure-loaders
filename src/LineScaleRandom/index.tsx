/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
