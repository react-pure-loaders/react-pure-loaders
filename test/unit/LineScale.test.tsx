import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import LineScale from '../../src/LineScale';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<LineScale>', () => {
    afterEach(cleanup);

    test('LineScale should match snapshot', () => {
        const { container } = render(<LineScale loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('LineScale should have default color', () => {
        const { container } = render(<LineScale loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('LineScale should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<LineScale color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('LineScale should have no children', () => {
        const { container } = render(<LineScale loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('LineScale should have three children', () => {
        const { container } = render(<LineScale loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(6);
    });
});
