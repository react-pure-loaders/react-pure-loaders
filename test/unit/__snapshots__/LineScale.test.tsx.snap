// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineScale> LineScale should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -moz-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -moz-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation: animation-0 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s -0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation: animation-0 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s -0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation: animation-0 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s -0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(4) {
  -webkit-animation: animation-0 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s -0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(5) {
  -webkit-animation: animation-0 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s 0.00s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(6) {
  -webkit-animation: animation-0 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s 0.10s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(7) {
  -webkit-animation: animation-0 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s 0.20s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(8) {
  -webkit-animation: animation-0 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s 0.30s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div:nth-of-type(9) {
  -webkit-animation: animation-0 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
  animation: animation-0 1s 0.40s infinite cubic-bezier(.2,.68,.18,1.08);
}

.emotion-0>div {
  border-radius: 2px;
  height: 35px;
  margin: 2px;
  width: 4px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
